src/rotation.zig

```diff
@@ -321,6 +321,190 @@ pub const RotationModes = struct {
+		}
+
+		pub fn generateData(
+			_: *main.game.World,
+			_: Vec3i,
+			_: Vec3f,
+			_: Vec3f,
+			_: Vec3i,
+			neighbor: ?Neighbor,
+			currentBlock: *Block,
+			neighborBlock: Block,
+			blockPlacing: bool,
+		) bool {
+			const blockBaseModel = blocks.meshes.modelIndexStart(currentBlock.*);
+			const neighborBaseModel = blocks.meshes.modelIndexStart(neighborBlock);
+
+			if(blockPlacing or (blockBaseModel == neighborBaseModel) or neighborBlock.solid()) {
+				const neighborModel = blocks.meshes.model(neighborBlock);
+
+				var currentData = BranchData.init(currentBlock.data);
+				// Branch block upon placement should extend towards a block it was placed
+				// on if the block is solid or also uses branch model.
+				const targetVal = (
+					(neighborBlock.solid() and !neighborBlock.viewThrough())
+					and ((blockBaseModel == neighborBaseModel) or main.models.models.items[neighborModel].isNeighborOccluded[neighbor.?.reverse().toInt()])
+				);
+				currentData.setConnection(neighbor.?, targetVal);
+
+				const result: u16 = currentData.enabledConnections;
+				if(result == currentBlock.data) return false;
+
+				currentBlock.data = result;
+				return true;
+			}
+			return false;
+		}
+
+		pub fn updateData(block: *Block, neighbor: Neighbor, neighborBlock: Block) bool {
+			const blockBaseModel = blocks.meshes.modelIndexStart(block.*);
+			const neighborBaseModel = blocks.meshes.modelIndexStart(neighborBlock);
+			var currentData = BranchData.init(block.data);
+
+			// Handle joining with other branches. While placed, branches extend in a
+			// opposite direction than they were placed from, effectively connecting
+			// to the block they were placed at.
+			if(blockBaseModel == neighborBaseModel) {
+				const neighborData = BranchData.init(neighborBlock.data);
+				currentData.setConnection(neighbor, neighborData.isConnected(neighbor.reverse()));
+			} else if(!neighborBlock.solid()) {
+				currentData.setConnection(neighbor, false);
+			}
+
+			const result: u16 = currentData.enabledConnections;
+			if(result == block.data) return false;
+
+			block.data = result;
+			return true;
+		}
+
+		fn closestRay(
+			block: Block,
+			relativePlayerPos: Vec3f,
+			playerDir: Vec3f
+		) ?u16 {
+			var resultIntersection: ?RayIntersectionResult = null;
+			var resultBitMask: ?u16 = null;
+			// Check intersection with central part of the branch block.
+			{
+				const modelIndex = blocks.meshes.modelIndexStart(block);
+				if(RotationMode.DefaultFunctions.rayModelIntersection(modelIndex, relativePlayerPos, playerDir)) |intersection| {
+					// We can't return here, as any intersection that accidentally passes
+					// through the central part would delete the whole block, even if player
+					// was aiming at the connection.
+					resultIntersection = intersection;
+					resultBitMask = 0;
+				}
+			}
+			// In case we did not intersect with the central part, check all the side
+			// connections.
```

Comment:
should be one line.

Reply:
What is the strict column limit enforced in this repository?

Reply:
Also, this comment is outdated and makes no sense with current implementation, i will remove it.

Reply:
I think line limits are a relic from the past. Nowadays we can display 200 characters or more on the screen.
I think the focus should be one writing readable code, and breaking a line at an arbitrary width does not help with that in my opinion.

Reply:
> Also, this comment is outdated and makes no sense with current implementation, i will remove it.

Yeah, that's one of the reasons, why I don't like seeing so many comments, they tend to get outdated soon.

Reply:
The fact that you can do something doesn't mean you should do it. Realistically, it is very handy to open two files side by side and with eg. my screen (4K) it imposes ~150 column limit before I have to scroll to see the whole line of both files. I might be too sensitive most of the time since I'm usually cutting at 88 or 110. If not a strict rule, a guideline should be there, to stick with X columns max unless doing otherwise improves readability.

Reply:
If you want a fixed number for your own then I'd say 200 maybe.